### ISSUE TITLE
Omnilight and SpotLight incorrect on X11 editor

### ISSUE THREAD

**Issue by EvanNibbe** [godotengine/godot#68200]
### Godot version

v4.0.beta.custom_build [754552b34]

### System information

Ubuntu 20.04.5 LTS (64 bit) AMD Ryzen 9 5900x 12-core, AMD Dimgrey_cavefish GPU is AMD 6600XT

### Issue description

Half of the light created by the Omnilight and Spotlight is lost (creating dark areas along the -Z axis with respect to the local placement of the omnilight or spotlight).
This line to the dark area does not change with any amount of rotation of the Omnilight or Spotlight. Translating the light towards larger (absolute value-wise) negative numbers on the Z axis uniformly moves this line. 

### Steps to reproduce

My build (as a matter of avoiding crashes in my large project from earlier alpha builds of Godot) involves first altering core/object/message_queue.cpp and commenting out all the calls to the statistics(); function (this is a function that doesn't actually do anything besides use up memory and report back to the developers).
Compile Godot on linux with "scons target=editor CXXFLAGS=-O3 use_lto=yes platform=linuxbsd"
Open a new scene as a Node3D, put in a MeshInstance3D, load the attached .obj file as the mesh.
Then add a material override to the mesh, add in some color texture to it.
Add in an Omnilight, then add in a Spotlight.
Make the lights bigger (range of 1000 meters), turn on shadows, move them around (above the mesh), then see the dark area that forms along the negative z axis with respect to the position of the light.

[hexagon_object.zip](https://github.com/godotengine/godot/files/9925471/hexagon_object.zip)

**Comment by Calinou** [comment]
I can't reproduce this on 4.0.beta 282e50ac8 (Linux, AMD Radeon RX 6900 XT).

This is the project I created from scratch: [test_hexagon_shadow.zip](https://github.com/godotengine/godot/files/9931192/test_hexagon_shadow.zip)

https://user-images.githubusercontent.com/180032/199779063-93ac7eff-39a0-4a6d-86a7-28949e0ad495.mp4


@EvanNibbe Please upload a **complete** [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

**Comment by EvanNibbe** [comment]
I see now that the problem didn't have to do with the lights, it may have had to do with breaking changes in how the shaders worked from Godot Alpha 11 to Godot beta. I realize now my ocean tiles in the game were also changed in their own way.

**Comment by testman42** [comment]
I encountered this as well, in 4.2-beta4.
Seems reproducible. Here is a minimal reproduction project:
[omnilight.zip](https://github.com/godotengine/godot/files/13256644/omnilight.zip)
Notice how omnilight throws light only on one side of the mesh called `problematic_mesh`
But I noticed that that mesh is strange, as in it is placed far away from it's 0,0,0.
By moving the node and rotating it 180°, I managed to find out that the omnilight now affects it in the opposite direction.
Check it by hiding the `problematic_mesh` and making the `problematic_mesh_rotated` visible.

So the my suspicion is  that lighting gets messed up on meshes that are placed far away from their root or what the 0, 0, 0 is called in coordinate system.
